python/tskit/trees.py

```diff
@@ -3038,6 +3038,84 @@ def simplify(
         else:
             return new_ts
 
+    def slice(
+            self, start=None, stop=None, rescale_coordinates=True, simplify=True,
+            record_provenance=True):
+        """
+        Truncate this tree sequence, returning a tree sequence that covers a reduced
+        genomic span. Sites not covered by the sliced region are thrown away.
+
+        :param float start: The leftmost genomic position, giving the (open) start point
+            of the kept region. Information along the genome prior to this point will
+            be discarded. If None, use the start position of the current tree sequence.
+        :param float stop: The rightmost genomic position, giving the (closed) end point
+            of the kept region. Information along the genome at and past this point will
+            be discarded. If None, use the end position of the current tree sequence.
+        :param bool rescale_coordinates: Rescale genomic coordinates such that position
+            0 in the new tree sequence corresponds to position ``start`` in the old one.
+            All sites and tree intervals will thus change to reflect their new positions.
+            If ``False``, do not rescale: the resulting tree sequence will therefore
+            cover the same genomic length as the original, but will have tree and site
+            information missing for genomic regions outside the sliced span.
+            (Default: True)
```

Comment:
I vote for not having this option at all, but if you do really want it, definately have the default as False. (Generally, I'd like to encourage people to use real coordinates; things are less confusing that way.)

Reply:
It seems a little confusing to have a tree sequence with missing trees for certain sections. When simulating, I'm most likely to rescale. However, with real data, I agree that keeping original coordinates is less confusing. I'm happy to go with `False` as the default, but I'd definitely want to keep the possibility of chopping out the middle of a TS and saving it as an entirely separate object with new coordinates.

Reply:
(I should say that this was the original motivation for coding up `slice`: I wanted to pick out the middle of a genomic segment I was using as a test case for `tsinfer`, and just focus on inferring that region. The coordinates in this case were irrelevant, and it would have been very confusing to plot test cases with empty regions to left and right of the focal region.)

Reply:
Oh, wait! Let me retract my statement - I read too quickly and thought `rescale` meant to make the new tree sequence have the same length as the old one. Apologies. Instead: I agree with this, but can we call it something else? Like `shift` or `shift_to_zero`?

Reply:
I see. Yes, "rescale" might imply changing the meaning of a unit as well as shifting along, so not such a good choice of wording. How about `shift_coordinates=True/False` instead?

Reply:
Or perhaps simply `reset_coordinates`

Reply:
Now renamed to to `reset_coordinates`, as that correctly implies shifting so that the leftmost tree is at 0.

Reply:
sounds good. thanks!